field/eisenstein/eisenstein.go

```diff
@@ -114,31 +110,59 @@ func (z *ComplexNumber) Norm() *big.Int {
 // Quo sets z to the quotient of x and y, and returns z.
 func (z *ComplexNumber) Quo(x, y *ComplexNumber) *ComplexNumber {
 	norm := y.Norm()
+	if norm.Cmp(big.NewInt(0)) == 0 {
+		panic("division by zero")
+	}
 	z.Conjugate(y)
 	z.Mul(x, z)
 	z.A0.Div(&z.A0, norm)
 	z.A1.Div(&z.A1, norm)
 	return z
 }
 
+// QuoRem sets z to the quotient of x and y, r to the remaind, and returns z and r.
+func (z *ComplexNumber) QuoRem(x, y, r *ComplexNumber) (*ComplexNumber, *ComplexNumber) {
+	norm := y.Norm()
+	if norm.Cmp(big.NewInt(0)) == 0 {
+		panic("division by zero")
+	}
+	z.Conjugate(y)
+	z.Mul(x, z)
+	z.A0.Div(&z.A0, norm)
+	z.A1.Div(&z.A1, norm)
+	var t ComplexNumber
+	r.Sub(x, t.Mul(y, z))
+	return z, r
+}
+
+// Min returns the minimum of z... with respect to the norm.
+func Min(z ...*ComplexNumber) *ComplexNumber {
+	min := z[0]
```

Comment:
If `len(z) == 0` then get out of bounds panic. Dunno what is the best approach, return nil? Or explicit panic that the number of arguments to `Min` is zero. At some point we end up with a panic anyway, but otherwise it may be difficult to figure out what was the actual reason.

Reply:
Removed this method as it is no longer used